logstash-core/lib/logstash/logging/slowlog_manager.rb

```diff
@@ -0,0 +1,50 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging/slow_logger"
+
+module LogStash; module Logging
+  class SlowLogManager
+
+    attr_reader :loggers
+
+    def initialize
+      @loggers = {}
+    end
+
+    def freq_items
+      items = {}
+      @loggers.each do |key, logger|
+        report = {}
+        logger.freq_items.top_k.each do |item|
+          _key = item[0]
+          _report = logger.freq_items.report(_key)
+          stats   = _report.delete(:statistics)
+          _report[:number_of_events] = item[1]
+          _report.merge!(stats.to_hash)
+          report[_key] = _report
+        end
+        items[key] = report
+      end
+      items
+    end
+
+    def top_items
```

Comment:
The TopItems class is a leftover from early experimentation, I keep it just in case it was interesting during review, but I will remove it 😄  as it might get you confused.